beacon-chain/state/stateutil/sync_committee.root.go

```diff
@@ -37,13 +39,24 @@ func SyncCommitteeRoot(committee *ethpb.SyncCommittee) ([32]byte, error) {
 	}
 	fieldRoots = [][32]byte{pubkeyRoot, aggregateKeyRoot}
 
-	return ssz.BitwiseMerkleizeArrays(hasher, fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
+	return ssz.BitwiseMerkleize(hasher, fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
 }
 
 func merkleizePubkey(hasher ssz.HashFn, pubkey []byte) ([32]byte, error) {
-	chunks, err := ssz.Pack([][]byte{pubkey})
+	chunks, err := ssz.PackByChunk([][]byte{pubkey})
 	if err != nil {
 		return [32]byte{}, err
 	}
-	return ssz.BitwiseMerkleize(hasher, chunks, uint64(len(chunks)), uint64(len(chunks)))
+	var pubKeyRoot [32]byte
+	if features.Get().EnableVectorizedHTR {
+		outputChunk := make([][32]byte, 1)
+		htr.VectorizedSha256(chunks, outputChunk)
+		pubKeyRoot = outputChunk[0]
```

Comment:
I suggest we export directly a helper function on gohashtree to avoid the allocation of a slice in these cases.